### PR TITLE
Add RGB matrix & LED Matrix support for IS31FL3742A, IS31FL3743A, IS31FL3745, IS31FL3746A

### PR DESCRIPTION
**Pull request by MasterSpoon**
Adding a ISSI Common driver that adds support for 4 newer drivers and sets up backbone to bring further drivers onboard through a more central process. 

Have added hooks for RGB and LED Matrix to allow selection of IS31FLCOMMON driver.
Added driver to allow swapping in config files for different ISSI models as all share common operation
Added config files for IS31FL3742A, IS31FL3743A, IS31FL3745, IS31FL3746A
Updated docs with how to configure new features

Have been able to test on hardware IS31FL3743A, the other three are assumed to be functional by association as they are operated in the same manner. Have done a dummy config and compile for all the 3 other drivers to ensure they compile correctly however have not been hardware tested at this stage.

I have used the existing configurations and drivers as examples of how to layout and configure to try keep my coded in line with the current coding in the repo.

Both the led_matrix_drivers.c and rgb_matrix_drivers.c have drivers that are the fall back if nothing else defined so have placed hooks for common driver before there.

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Has some merge conflicts

**Comment by MasterSpoon** [comment]
Updated conflicts and formatted in line with #14760 (which is much nicer then it was by the way, thank you fauxpark)

**Comment by drashna** [comment]
Discord discussion about this, starting here:
<https://discord.com/channels/440868230475677696/867530715192885269/904495820995432448>

for future drashna's reference

**Comment by drashna** [comment]
This has some merge conflicts that need to be resolved.

**Comment by MasterSpoon** [comment]
Resolved merge conflicts.

**Review by drashna — APPROVED** [review]

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -232,6 +232,115 @@ const is31_led __flash g_is31_leds[DRIVER_LED_TOTAL] = {
...
 Where `X_Y` is the location of the LED in the matrix defined by [the datasheet](https://www.issi.com/WW/pdf/31FL3737.pdf) and the header file `drivers/led/issi/is31fl3737.h`. The `driver` is the index of the driver you defined in your `config.h` (Only `0`, `1` for now).
 
+---
+### IS31FLCOMMON :id=is31flcommon
+
+There is basic support for addressable RGB matrix lighting with a selection of I2C ISSI Lumissil RGB controllers through a shared common driver. To enable it, add this to your `rules.mk`:
+
+```makefile
+LED_MATRIX_ENABLE = yes
+LED_MATRIX_DRIVER = <driver name>
```
Comment (Xelus22) [inline docs/feature_rgb_matrix.md:242]:
```suggestion
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = <driver name>
```

**Review by noroadsleft — APPROVED** [review]
Not super familiar with the relevant features, but I don't see anything concerning here.

`__attribute__ ((weak))` :heavy_check_mark:

**Pull request merged**